src/game/scripts/vscripts/abilities/pudge_flesh_heap_health_regeneration.lua

```diff
@@ -0,0 +1,143 @@
+pudge_flesh_heap_health_regeneration = class({})
+
+LinkLuaModifier("modifier_flesh_heap_health_regeneration", "abilities/pudge_flesh_heap_health_regeneration.lua" ,LUA_MODIFIER_MOTION_NONE )
+
+
+function pudge_flesh_heap_health_regeneration:GetIntrinsicModifierName()
+  return "modifier_flesh_heap_health_regeneration"
+end
+
+modifier_flesh_heap_health_regeneration = class({})
+
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:IsHidden()
+    if self:GetAbility():GetLevel() == 0 then
+        return true
+    end
+    return false
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:RemoveOnDeath()
+    return false
+end
+
+--------------------------------------------------------------------------------
+function modifier_flesh_heap_health_regeneration:IsPurgable()
+    return false
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:IsPassive()
+    return true
+end
+
+function modifier_flesh_heap_health_regeneration:IsPurgable()
+    return true
+end
+
+function modifier_flesh_heap_health_regeneration:GetFleshHeapKills()
+  if self.nKills == nil then
+    self.nKills = 0
+  end
+  return self.nKills
+end
+ 
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:OnCreated( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_health_regeneration")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:OnRefresh( kv )
```

Comment:
You are right about this one, it is because it used to be based on the ability function OnNearbyHeroKilled but that had limitations. It really doesn't matter though.